src/lualdap.c

```diff
@@ -1008,20 +1008,18 @@ static int lualdap_open_simple (lua_State *L) {
 	conn_data *conn = (conn_data *)lua_newuserdata (L, sizeof(conn_data));
 #if defined(LDAP_API_FEATURE_X_OPENLDAP) && LDAP_API_FEATURE_X_OPENLDAP >= 20300
 	struct berval *cred = NULL;
-	char *host_with_schema = NULL;
+	const char *host_with_schema = NULL;
 #endif
 	int err;
 
 	/* Initialize */
 	lualdap_setmeta (L, LUALDAP_CONNECTION_METATABLE);
 	conn->version = 0;
 #if defined(LDAP_API_FEATURE_X_OPENLDAP) && LDAP_API_FEATURE_X_OPENLDAP >= 20300
-	host_with_schema = malloc(strlen(host) + 8);
-	strcpy(host_with_schema, "ldap://");
-	strcat(host_with_schema, host);
+	host_with_schema = lua_pushfstring(L, "ldap://%s", host);
```

Comment:
I like how this reduces noise. Please rebase onto ae98ecfaf59b7cbd01c99d7ebce1a9ac3ccc6403.